server/py/services/api/tests/unit/db/test_artifacts.py

```diff
@@ -1040,6 +1040,203 @@ def test_delete_artifact_tag_filter(self):
         tags = self._db.list_artifact_tags(self._db_session, project)
         assert len(tags) == 0
 
+    def test_delete_artifact_with_latest_tag_and_iteration_0(
+        self,
+    ):
+        # This test is based on the following scenario:
+        # 1. Log an artifact hyperparameters - iteration 1, best_iteration=True
+        # 2. Log an artifact hyperparameters - iteration 2, best_iteration=False
+        # 3. Log an artifact without hyperparameters - iteration 0, best_iteration=True
+        # 4. Delete the artifact with the "latest" tag - the last artifact that was logged (iteration 0)
+        # 5. The "latest" tag should move to the most recent artifact with the best_iteration flag
+        # (the first artifact that was logged)
+
+        project = "artifact_project"
+        artifact_key = "artifact-key"
+        artifact_1_tree = "artifact_tree_1"
+        artifact_1_body = self._generate_artifact(artifact_key, tree=artifact_1_tree)
+        artifact_2_tree = "artifact_tree_2"
+        artifact_2_body = self._generate_artifact(artifact_key, tree=artifact_2_tree)
+
+        # Log the first artifact as part of a function run with hyperparameters (iteration 1, best_iteration=True)
+        uid1 = self._db.store_artifact(
+            self._db_session,
+            artifact_key,
+            artifact_1_body,
+            project=project,
+            tag="v1",
+            iter=1,
+            best_iteration=True,
+        )
+
+        # Log the second artifact as part of a function run with hyperparameters (iteration 2, without best_iteration)
+        uid2 = self._db.store_artifact(
+            self._db_session,
+            artifact_key,
+            artifact_1_body,
+            project=project,
+            tag="v2",
+            iter=2,
+        )
+
+        # Logging artifact in a regular way, with iteration = 0 best_iteration=True.
+        # The "latest" tag should now be assigned to this artifact.
+        uid3 = self._db.store_artifact(
+            self._db_session,
+            artifact_key,
+            artifact_2_body,
+            project=project,
+            tag="v3",
+            best_iteration=True,
+        )
+
+        assert uid1 != uid2 != uid3
+        artifacts = self._db.list_artifacts(
+            self._db_session, name=artifact_key, project=project
+        )
+        assert len(artifacts) == 4
```

Comment:
Why 4? I'm not really familiar with how we log artifacts, may you add a comment explaining that?